operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -53,20 +53,30 @@ func CreateOrUpdateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client
 	key := client.ObjectKey{Name: stack.Spec.Storage.Secret.Name, Namespace: stack.Namespace}
 	if err := k.Get(ctx, key, &s3secret); err != nil {
 		if apierrors.IsNotFound(err) {
-			return status.SetDegradedCondition(ctx, k, req,
+			statusErr := status.SetDegradedCondition(ctx, k, req,
 				"Missing object storage secret",
 				lokiv1beta1.ReasonMissingObjectStorageSecret,
 			)
+
+			if statusErr == nil {
+				return kverrors.Wrap(err, "lokistack s3 secret has not been created", "name", key)
+			}
+			return kverrors.Wrap(statusErr, "failed to update lokistack condition with missing secret status")
```

Comment:
This works fine, but the following code looks more like canonical Go code to me:
```suggestion
			if statusErr := status.SetDegradedCondition(ctx, k, req,
				"Missing object storage secret",
				lokiv1beta1.ReasonMissingObjectStorageSecret,
			); statusErr != nil {
				return kverrors.Wrap(statusErr, "failed to update lokistack condition with missing secret status")
			}

			return kverrors.Wrap(err, "lokistack s3 secret has not been created", "name", key)
```

It makes it clear that the `statusErr` is an additional error condition overlaying the other one and the original error will be returned in any case. Because the `if` is an additional scope you can also replace `statusErr` with `err` and it will shadow the original error inside the if-block then.

What do you think?